TAF/Tests/SearchTests.cs

```diff
@@ -0,0 +1,223 @@
+        private const string _allFields = "\\SearchQueryWithAllFieldsFilledInJobListingsPageTest.json";
+        private const string _unsuccessfulQuery = "\\UnsuccessfulSearchJobListingsPageTest.json";
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mainPage = new();
+            _searchResultsPage = new();
+            _jobListingsPage = new();
+        }
+
+        [NonParallelizable]
+        [TestCase("Business Analysis")]
+        public void SearchPanelResultIsCorrectTest(string expectedSearchResult)
+        {
+            _mainPage.Header.SearchButton.Click();
+            _mainPage.Header.SearchForm.SendKeys(expectedSearchResult);
+            _mainPage.Header.FindResultButton.Click();
+
+            Waiters.WaitForCondition(new Func<bool>(() => _searchResultsPage.SearchFormOnResultPage.IsDisplayed()));
+
+            Assert.That(_searchResultsPage.SearchFormOnResultPage.GetProperty("value"), Is.EqualTo(expectedSearchResult), "Search results don't match");
+        }
+
+        [NonParallelizable]
+        [TestCase(5, "Automation")]
+        public void EnteredTextIsPresentInSearchResultsTest(int expectedNumberOfResults, string searchQuery)
+        {
+            _mainPage.Header.SearchButton.Click();
+            _mainPage.Header.SearchForm.SendKeys(searchQuery);
+            _mainPage.Header.FindResultButton.Click();
+
+            Assert.That(_searchResultsPage.ListOfArticles.GetElements().Take(expectedNumberOfResults).All(p => p.GetProperty("textContent").ToLower().Contains(searchQuery.ToLower())), Is.True, "First five articles don't contains search query");
+        }
+
+        [NonParallelizable]
+        [TestCase("Business Analysis")]
+        public void ArticleTitleMatchTest(string expectedSearchResult)
+        {
+            _mainPage.Header.SearchButton.Click();
+            _mainPage.Header.SearchForm.SendKeys(expectedSearchResult);
+            _mainPage.Header.FindResultButton.Click();
+
+            Waiters.WaitForCondition(new Func<bool>(() => _searchResultsPage.FirstArticle.IsDisplayed()));
+
+            var firstSearchResultTitle = _searchResultsPage.FirstArticle.GetAttribute("innerText");
+            _searchResultsPage.FirstArticle.Click();
+
+            Assert.That(firstSearchResultTitle, Is.EqualTo(_searchResultsPage.FirstSearchResultTitle.GetProperty("innerText")), "Title of the first search result don't match with title of the first article");
+        }
+
+        [NonParallelizable]
+        [TestCase("Automation", "20")]
+        public void QuantityOfSearchResultsTest(string searchQuery, string expectedQuantityOfArticles)
+        {
+            _mainPage.Header.SearchButton.ClickUsingJS();
+            _mainPage.Header.SearchForm.SendKeys(searchQuery);
+            _mainPage.Header.FindResultButton.Click();
+
+            Browser.NewBrowser.ScrollToElement(_searchResultsPage.ListOfArticles.GetElements()[9].OriginalWebElement);
```

Comment:
It is not a good idea to scroll to the define element because list can return less than 9 elements and test will fail. So change the element to be scrolled.

Reply:
Resolved